docs/tutorials/on-device-training/ios-app.md

```diff
@@ -0,0 +1,991 @@
+                      dynamic_axes={"input": {0: "batch"}, "output": {0: "batch"}})
+    ```
+2. ### Define the trainable and non trainable parameters
+
+    ```python
+    import onnx
+
+    # load the onnx model
+    onnx_model = onnx.load("wav2vec.onnx")
+
+    # Define the parameters that require gradients to be computed (trainable parameters) and
+    # those that don't (frozen/non-trainable parameters)
+    requires_grad = ["classifier.weight", "classifier.bias"]
+    frozen_params = [
+        param.name
+        for param in onnx_model.graph.initializer
+        if param.name not in requires_grad
+    ]
+    ```
+3. ### Generate the training artifacts. 
+    We will use the `CrossEntropyLoss` loss and the `AdamW` optimizer for this tutorial. More details around artifact generation can be found [here](../../../docs/api/python/on_device_training/training_artifacts.html).
+
+    Since the model also outputs logits and hidden states, we will use `onnxblock` to define a custom loss function that extracts logits from the model output and passes it to the `CrossEntropyLoss` function.
+
+    ```python
+    import onnxruntime.training.onnxblock as onnxblock
+    from onnxruntime.training import artifacts
+
+    # define the loss function
+    class CustomCELoss(onnxblock.Block):
+        def __init__(self):
+            super().__init__()
+            self.celoss = onnxblock.loss.CrossEntropyLoss()
+
+
+        def build(self, logits, *args):
+            return self.celoss(logits)
+
+
+    # Generate the training artifacts
+    artifacts.generate_artifacts(
+        onnx_model,
+        requires_grad=requires_grad,
+        frozen_params=frozen_params,
+        loss=CustomCELoss(),
+        optimizer=artifacts.OptimType.AdamW,
+        artifact_directory="artifacts",
+    )
+
+    ```
+    That's all! The training artifacts have been generated in the `artifacts` directory. The artifacts are ready to be deployed to the iOS device for training.
+
+
+## Building the iOS application
+### Xcode Setup
+Open Xcode, and create a new project. Select `iOS` as the platform and `App` as the template. Click Next.
+
+![Xcode Setup New Project](../../../images/iOS_speaker_identification_xcode_new_app_screen.png)
+
+Enter the project name. Here, we will name the project 'MyVoice', but you can name it anything you prefer. Make sure to select `SwiftUI` as the interface, and `Swift` as the language. Then, click Next.
+
+![Xcode Setup Project Name](../../../images/iOS_speaker_identification_xcode_new_project_screen.png)
+
+Select the location where you want to save the project and click `Create`.
+
+
+Now, we will need to add the `onnxruntime-training-objc` pods to the project. We will use `CocoaPods` to add the dependency. If you don't have CocoaPods installed, you can install it by running the following command in the terminal:
+
+```bash
+sudo gem install cocoapods
+```
```

Comment:
Instead of providing cocoapods installation instructions, can we share the link to where they can refer to for installation instructions?